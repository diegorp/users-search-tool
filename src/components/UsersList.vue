<template>
    <template v-for="user in users" :key="user.id">
        <UserInfo :user="user"/>
    </template>
</template>

<script>
import UserInfo from './UserInfo.vue'

export default {
    name: 'UsersList',
    props: {
        users: Array
    },
    components: {
        UserInfo
    },
}
</script>

<style scoped lang="scss">
    .user:not(:first-child) {
        margin-top: 2px;
    }
</style>