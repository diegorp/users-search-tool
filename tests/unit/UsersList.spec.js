import { shallowMount } from "@vue/test-utils";
import UsersList from '@/components/UsersList.vue'

describe('UsersList.vue', () => {
    let wrapper;
    const user = {
        id: 9,
        name: 'John Doe',
        email: 'johndoe@mail.com',
        phone: '(775)976-6794 x41206',
        company: {
            name: 'Alkimii',
            catchPhrase: 'Integrated Applications made by our hotel people for your hotel people.'
        }
    }

    beforeEach(() => {
        wrapper = shallowMount(UsersList, {
            props: { users: [user, user] }
        });
    });

    it('should render the component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});