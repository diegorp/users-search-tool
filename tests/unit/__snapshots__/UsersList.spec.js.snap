// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList.vue should render the component 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      
      <user-info-stub
        user="[object Object]"
      />
      <user-info-stub
        user="[object Object]"
      />
      
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "hasOwnProperty": [Function],
          "users": Array [
            Object {
              "company": Object {
                "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                "name": "Alkimii",
              },
              "email": "johndoe@mail.com",
              "id": 9,
              "name": "John Doe",
              "phone": "(775)976-6794 x41206",
            },
            Object {
              "company": Object {
                "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                "name": "Alkimii",
              },
              "email": "johndoe@mail.com",
              "id": 9,
              "name": "John Doe",
              "phone": "(775)976-6794 x41206",
            },
          ],
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "isVueInstance": 0,
            "toJSON": 0,
            "users": 3,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {
            "UserInfo": Object {
              "name": "UserInfo",
              "props": Object {
                "user": [Function],
              },
              "render": [Function],
            },
          },
          "ctx": Object {
            "hasOwnProperty": [Function],
            "users": Array [
              Object {
                "company": Object {
                  "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                  "name": "Alkimii",
                },
                "email": "johndoe@mail.com",
                "id": 9,
                "name": "John Doe",
                "phone": "(775)976-6794 x41206",
              },
              Object {
                "company": Object {
                  "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                  "name": "Alkimii",
                },
                "email": "johndoe@mail.com",
                "id": 9,
                "name": "John Doe",
                "phone": "(775)976-6794 x41206",
              },
            ],
          },
          "da": null,
          "data": Object {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "users": Array [
              Object {
                "company": Object {
                  "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                  "name": "Alkimii",
                },
                "email": "johndoe@mail.com",
                "id": 9,
                "name": "John Doe",
                "phone": "(775)976-6794 x41206",
              },
              Object {
                "company": Object {
                  "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                  "name": "Alkimii",
                },
                "email": "johndoe@mail.com",
                "id": 9,
                "name": "John Doe",
                "phone": "(775)976-6794 x41206",
              },
            ],
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "users": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "hasOwnProperty": [Function],
            "users": Array [
              Object {
                "company": Object {
                  "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                  "name": "Alkimii",
                },
                "email": "johndoe@mail.com",
                "id": 9,
                "name": "John Doe",
                "phone": "(775)976-6794 x41206",
              },
              Object {
                "company": Object {
                  "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                  "name": "Alkimii",
                },
                "email": "johndoe@mail.com",
                "id": 9,
                "name": "John Doe",
                "phone": "(775)976-6794 x41206",
              },
            ],
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": ,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": Object {
                  "a": null,
                  "accessCache": Object {},
                  "appContext": Object {
                    "app": [Circular],
                    "components": Object {
                      "transition": Object {
                        "name": "transition",
                      },
                      "transition-group": Object {
                        "name": "transition-group",
                      },
                    },
                    "config": Object {
                      "compilerOptions": Object {},
                      "errorHandler": undefined,
                      "globalProperties": Object {},
                      "isNativeTag": [Function],
                      "optionMergeStrategies": Object {},
                      "performance": false,
                      "warnHandler": undefined,
                    },
                    "directives": Object {},
                    "emitsCache": WeakMap {},
                    "mixins": Array [
                      Object {
                        "beforeCreate": [Function],
                      },
                    ],
                    "optionsCache": WeakMap {},
                    "propsCache": WeakMap {},
                    "provides": Object {},
                    "reload": [Function],
                  },
                  "asyncDep": null,
                  "asyncResolved": false,
                  "attrs": Object {},
                  "bc": null,
                  "bm": null,
                  "bu": null,
                  "bum": null,
                  "c": null,
                  "components": null,
                  "ctx": Object {
                    "user": Object {
                      "company": Object {
                        "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                        "name": "Alkimii",
                      },
                      "email": "johndoe@mail.com",
                      "id": 9,
                      "name": "John Doe",
                      "phone": "(775)976-6794 x41206",
                    },
                  },
                  "da": null,
                  "data": Object {},
                  "directives": null,
                  "ec": null,
                  "effect": ReactiveEffect {
                    "active": true,
                    "allowRecurse": true,
                    "deps": Array [
                      Set {
                        [Circular],
                      },
                      Set {
                        [Circular],
                      },
                    ],
                    "fn": [Function],
                    "onTrack": undefined,
                    "onTrigger": undefined,
                    "parent": undefined,
                    "scheduler": [Function],
                  },
                  "emit": [Function],
                  "emitsOptions": null,
                  "emitted": null,
                  "exposeProxy": null,
                  "exposed": null,
                  "inheritAttrs": undefined,
                  "isDeactivated": false,
                  "isMounted": true,
                  "isUnmounted": false,
                  "m": null,
                  "next": null,
                  "parent": [Circular],
                  "props": Object {
                    "user": Object {
                      "company": Object {
                        "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                        "name": "Alkimii",
                      },
                      "email": "johndoe@mail.com",
                      "id": 9,
                      "name": "John Doe",
                      "phone": "(775)976-6794 x41206",
                    },
                  },
                  "propsDefaults": Object {},
                  "propsOptions": Array [
                    Object {
                      "user": Object {
                        "0": false,
                        "1": true,
                        "type": [Function],
                      },
                    },
                    Array [],
                  ],
                  "provides": Object {},
                  "proxy": Object {
                    "user": Object {
                      "company": Object {
                        "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                        "name": "Alkimii",
                      },
                      "email": "johndoe@mail.com",
                      "id": 9,
                      "name": "John Doe",
                      "phone": "(775)976-6794 x41206",
                    },
                  },
                  "refs": Object {},
                  "render": [Function],
                  "renderCache": Array [],
                  "root": [Circular],
                  "rtc": null,
                  "rtg": null,
                  "scope": EffectScope {
                    "active": true,
                    "cleanups": Array [],
                    "effects": Array [
                      ReactiveEffect {
                        "active": true,
                        "allowRecurse": true,
                        "deps": Array [
                          Set {
                            [Circular],
                          },
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                    ],
                  },
                  "setupContext": Object {
                    "attrs": Object {},
                    "emit": [Function],
                    "expose": [Function],
                    "slots": Object {},
                  },
                  "setupState": Object {},
                  "slots": Object {},
                  "sp": null,
                  "subTree": Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <user-info-stub
                      user="[object Object]"
                    />,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "user": Object {
                        "company": Object {
                          "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                          "name": "Alkimii",
                        },
                        "email": "johndoe@mail.com",
                        "id": 9,
                        "name": "John Doe",
                        "phone": "(775)976-6794 x41206",
                      },
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "user-info-stub",
                  },
                  "suspense": null,
                  "suspenseId": 0,
                  "type": Object {
                    "name": "UserInfo",
                    "props": Object {
                      "user": [Function],
                    },
                    "setup": [Function],
                  },
                  "u": null,
                  "uid": 2,
                  "um": null,
                  "update": [Function],
                  "vnode": [Circular],
                  "withProxy": null,
                },
                "dirs": null,
                "dynamicChildren": Array [
                  [Circular],
                ],
                "dynamicProps": Array [
                  "user",
                ],
                "el": <user-info-stub
                  user="[object Object]"
                />,
                "key": 9,
                "patchFlag": 8,
                "props": Object {
                  "key": 9,
                  "user": Object {
                    "company": Object {
                      "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                      "name": "Alkimii",
                    },
                    "email": "johndoe@mail.com",
                    "id": 9,
                    "name": "John Doe",
                    "phone": "(775)976-6794 x41206",
                  },
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 4,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Object {
                  "name": "UserInfo",
                  "props": Object {
                    "user": [Function],
                  },
                  "setup": [Function],
                },
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": Object {
                  "a": null,
                  "accessCache": Object {},
                  "appContext": Object {
                    "app": [Circular],
                    "components": Object {
                      "transition": Object {
                        "name": "transition",
                      },
                      "transition-group": Object {
                        "name": "transition-group",
                      },
                    },
                    "config": Object {
                      "compilerOptions": Object {},
                      "errorHandler": undefined,
                      "globalProperties": Object {},
                      "isNativeTag": [Function],
                      "optionMergeStrategies": Object {},
                      "performance": false,
                      "warnHandler": undefined,
                    },
                    "directives": Object {},
                    "emitsCache": WeakMap {},
                    "mixins": Array [
                      Object {
                        "beforeCreate": [Function],
                      },
                    ],
                    "optionsCache": WeakMap {},
                    "propsCache": WeakMap {},
                    "provides": Object {},
                    "reload": [Function],
                  },
                  "asyncDep": null,
                  "asyncResolved": false,
                  "attrs": Object {},
                  "bc": null,
                  "bm": null,
                  "bu": null,
                  "bum": null,
                  "c": null,
                  "components": null,
                  "ctx": Object {
                    "user": Object {
                      "company": Object {
                        "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                        "name": "Alkimii",
                      },
                      "email": "johndoe@mail.com",
                      "id": 9,
                      "name": "John Doe",
                      "phone": "(775)976-6794 x41206",
                    },
                  },
                  "da": null,
                  "data": Object {},
                  "directives": null,
                  "ec": null,
                  "effect": ReactiveEffect {
                    "active": true,
                    "allowRecurse": true,
                    "deps": Array [
                      Set {
                        [Circular],
                      },
                      Set {
                        [Circular],
                      },
                    ],
                    "fn": [Function],
                    "onTrack": undefined,
                    "onTrigger": undefined,
                    "parent": undefined,
                    "scheduler": [Function],
                  },
                  "emit": [Function],
                  "emitsOptions": null,
                  "emitted": null,
                  "exposeProxy": null,
                  "exposed": null,
                  "inheritAttrs": undefined,
                  "isDeactivated": false,
                  "isMounted": true,
                  "isUnmounted": false,
                  "m": null,
                  "next": null,
                  "parent": [Circular],
                  "props": Object {
                    "user": Object {
                      "company": Object {
                        "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                        "name": "Alkimii",
                      },
                      "email": "johndoe@mail.com",
                      "id": 9,
                      "name": "John Doe",
                      "phone": "(775)976-6794 x41206",
                    },
                  },
                  "propsDefaults": Object {},
                  "propsOptions": Array [
                    Object {
                      "user": Object {
                        "0": false,
                        "1": true,
                        "type": [Function],
                      },
                    },
                    Array [],
                  ],
                  "provides": Object {},
                  "proxy": Object {
                    "user": Object {
                      "company": Object {
                        "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                        "name": "Alkimii",
                      },
                      "email": "johndoe@mail.com",
                      "id": 9,
                      "name": "John Doe",
                      "phone": "(775)976-6794 x41206",
                    },
                  },
                  "refs": Object {},
                  "render": [Function],
                  "renderCache": Array [],
                  "root": [Circular],
                  "rtc": null,
                  "rtg": null,
                  "scope": EffectScope {
                    "active": true,
                    "cleanups": Array [],
                    "effects": Array [
                      ReactiveEffect {
                        "active": true,
                        "allowRecurse": true,
                        "deps": Array [
                          Set {
                            [Circular],
                          },
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                    ],
                  },
                  "setupContext": Object {
                    "attrs": Object {},
                    "emit": [Function],
                    "expose": [Function],
                    "slots": Object {},
                  },
                  "setupState": Object {},
                  "slots": Object {},
                  "sp": null,
                  "subTree": Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <user-info-stub
                      user="[object Object]"
                    />,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "user": Object {
                        "company": Object {
                          "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                          "name": "Alkimii",
                        },
                        "email": "johndoe@mail.com",
                        "id": 9,
                        "name": "John Doe",
                        "phone": "(775)976-6794 x41206",
                      },
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "user-info-stub",
                  },
                  "suspense": null,
                  "suspenseId": 0,
                  "type": Object {
                    "name": "UserInfo",
                    "props": Object {
                      "user": [Function],
                    },
                    "setup": [Function],
                  },
                  "u": null,
                  "uid": 3,
                  "um": null,
                  "update": [Function],
                  "vnode": [Circular],
                  "withProxy": null,
                },
                "dirs": null,
                "dynamicChildren": Array [
                  [Circular],
                ],
                "dynamicProps": Array [
                  "user",
                ],
                "el": <user-info-stub
                  user="[object Object]"
                />,
                "key": 9,
                "patchFlag": 8,
                "props": Object {
                  "key": 9,
                  "user": Object {
                    "company": Object {
                      "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                      "name": "Alkimii",
                    },
                    "email": "johndoe@mail.com",
                    "id": 9,
                    "name": "John Doe",
                    "phone": "(775)976-6794 x41206",
                  },
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 4,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Object {
                  "name": "UserInfo",
                  "props": Object {
                    "user": [Function],
                  },
                  "setup": [Function],
                },
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [],
            "dynamicProps": null,
            "el": ,
            "key": null,
            "patchFlag": 128,
            "props": null,
            "ref": null,
            "scopeId": null,
            "shapeFlag": 16,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": Symbol(Fragment),
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {
              "UserInfo": Object {
                "name": "UserInfo",
                "props": Object {
                  "user": [Function],
                },
                "render": [Function],
              },
            },
            "name": "UsersList",
            "props": Object {
              "users": [Function],
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
          "users": Array [
            Object {
              "company": Object {
                "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                "name": "Alkimii",
              },
              "email": "johndoe@mail.com",
              "id": 9,
              "name": "John Doe",
              "phone": "(775)976-6794 x41206",
            },
            Object {
              "company": Object {
                "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
                "name": "Alkimii",
              },
              "email": "johndoe@mail.com",
              "id": 9,
              "name": "John Doe",
              "phone": "(775)976-6794 x41206",
            },
          ],
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {
            "UserInfo": Object {
              "name": "UserInfo",
              "props": Object {
                "user": [Function],
              },
              "render": [Function],
            },
          },
          "name": "UsersList",
          "props": Object {
            "users": [Function],
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.33",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
    "users": Array [
      Object {
        "company": Object {
          "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
          "name": "Alkimii",
        },
        "email": "johndoe@mail.com",
        "id": 9,
        "name": "John Doe",
        "phone": "(775)976-6794 x41206",
      },
      Object {
        "company": Object {
          "catchPhrase": "Integrated Applications made by our hotel people for your hotel people.",
          "name": "Alkimii",
        },
        "email": "johndoe@mail.com",
        "id": 9,
        "name": "John Doe",
        "phone": "(775)976-6794 x41206",
      },
    ],
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": ,
}
`;
